src/shared/constants/web3.constants.ts

```diff
@@ -120,13 +120,29 @@ export const tokenLabels: ValueByTokenAndNetwork = {
 
 // @TODO Abstract a bit more
 export const buyTokenUrlByTokenAndNetwork: ValueByTokenAndNetwork = {
```

Comment:
Lots of string repetition here. If just the domain changed for Quickswap, we have 7 changes Maybe we can create util function in web3.util?
Something like `getDexUrl(ESupportedToken, Network)` and call it here in the map

Reply:
So instead of calling the getDexUrl in the object map. I just completely abstracted the dexUrl generation in that method since calling it in the constants won't work due to cyclic dependency. Handling will depend on the asset type. LP will still have an object map since each LP might come from different dexes. Tokens will just have options depending on the EChainId